src/Nancy/HttpLinkRelation.cs

```diff
@@ -0,0 +1,180 @@
+﻿namespace Nancy
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// The relation that identifies the semantics of a <see cref="HttpLink"/> contained in an RFC 5899 Link HTTP header,
+    /// as built by the <see cref="HttpLinkBuilder"/>.
+    /// </summary>
+    /// <seealso cref="System.Uri" />
+    [Serializable]
+    public class HttpLinkRelation : Uri, IEquatable<HttpLinkRelation>
+    {
+        /// <summary>
+        /// The URI prefix to use for IANA registered link relations.
+        /// </summary>
+        public static readonly Uri IanaLinkRelationPrefix = new Uri("http://www.iana.org/assignments/relation/");
+
+        private readonly Uri prefix;
+
+        private readonly string value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLinkRelation"/> class.
+        /// </summary>
+        /// <param name="relation">The relation.</param>
+        public HttpLinkRelation(string relation)
+            : this(Parse(relation).Prefix, Parse(relation).Value)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLinkRelation" /> class.
+        /// </summary>
+        /// <param name="prefix">The prefix.</param>
+        /// <param name="value">The value.</param>
+        public HttpLinkRelation(Uri prefix, string value)
+            : base(prefix + value)
+        {
+            this.prefix = prefix;
+            this.value = value;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLinkRelation"/> class.
+        /// </summary>
+        /// <param name="serializationInfo">An instance of the <see cref="T:System.Runtime.Serialization.SerializationInfo" /> class containing the information required to serialize the new <see cref="T:System.Uri" /> instance.</param>
+        /// <param name="streamingContext">An instance of the <see cref="T:System.Runtime.Serialization.StreamingContext" /> class containing the source of the serialized stream associated with the new <see cref="T:System.Uri" /> instance.</param>
+        protected HttpLinkRelation(SerializationInfo serializationInfo, StreamingContext streamingContext)
```

Comment:
This (and the `[Serializable]` attribute above) won't fly on .NET Core. I'm not really sure if we should keep this for now or just kill it sooner rather than later.